### ISSUE TITLE
Tagbar sets ballooneval globally on init

### ISSUE THREAD

**Issue by mmrwoods** [preservim/tagbar#499]
Hi,

Thanks for Tagbar, I find it really useful :-)

I've added some horrible little hacks to my vimrc to deal with the fact that it sets ballooneval globally when setting up the window though (i.e. `set ballooneval` rather than `setlocal ballooneval`).

Is there a reason why ballooneval is set using `set` rather than `setlocal`?

Thanks

Mark

**Comment by alerque** [comment]
No I don't think so, this should definitely be fixed so no global settings are changed. Would you like to contribute this as a PR?

**Comment by alerque** [comment]
In the mean time this entire feature seems to have been stripped from Neovim. I'm wondering if we should be including it at all in this plugin.

**Comment by alerque** [comment]
I'd be happy to make the one line change myself, but I'm not sure how to test it as I don't use VIM and don't encounter anywhere this would actually be used. If you can make the change and report that the plugin still works while not setting it globally I'd love to see that done here.

**Comment by mmrwoods** [comment]
Hi, sorry I'm just following up on this now. 

My memory is fuzzy at this stage, but I think the reason this was so much of a problem was one of the ruby related filetype plugins use of ballooneval was painfully slow on large files. It no longer seems to be a problem in vim 8.1.2234

I see a new option has been added to Tagbar to allow ballooneval to be disabled, `g:tagbar_show_balloon`. This is great, it solves the initial problem. Netrw has a similar `g:netrw_nobeval` option which disables ballooneval entirely in Netrw.

It would be nice to be able to enable ballooneval for just the Tagbar window though. I solved this in my `.vimrc` with a couple of fugly autocmds and a custom mapping to open Tagbar:

```
Workaround for tagbar setting ballooneval globally on open/toggle
if has('balloon_eval')
  autocmd WinEnter __Tagbar__* set ballooneval
  autocmd WinLeave __Tagbar__* set noballooneval
endif

" Outline view of current file
" Note: tagbar plugin automatically enables ballooneval globally
" nnoremap <leader>o :TagbarToggle<Enter>:set noballooneval<CR>
```

I see Netrw since version 143 solves this problem by keeping track of the initial value of `&beval` and using autocmds to set it back to the initial value when leaving the window. See https://github.com/vim/vim/blob/master/runtime/autoload/netrw.vim#L557

Maybe Tagbar would benefit from similar functionality? I can whip up a PR for this if you think it's a good idea.
